weekDownloader.py

```diff
@@ -62,15 +66,51 @@ def main(argv):
     all_units = get_unit_list(get_all_weeks(course_data), token)
     all_steps = get_steps_list(all_units, int(argv[1]), token)
     only_video_steps = get_only_video_steps(all_steps, token)
-    url_list_with_q = [x['video']['urls'][0] for x in only_video_steps]
+
+    url_list_with_q = []
+
+    for video_step in only_video_steps:
+        video_link = None
+
+        # Check a video quality.
+        for url in video_step['video']['urls']:
+            if url['quality'] == argv[3]:
+                video_link = url
+
+        # If the is no required video quality then download
+        # with the best available quality.
+        if video_link is None:
+            print("The requested quality = {} is not available!".format(argv[2]))
```

Comment:
at first we print a lot of info, and then trying to download ?
Looks strange to me:
<img width="671" alt="screen shot 2016-08-02 at 12 26 49 am" src="https://cloud.githubusercontent.com/assets/5033274/17309506/347a525a-5850-11e6-9853-ee58d6c2fed6.png">

It's better either to pass all data to video_link list as dicts (`video_link = [{'url': ..., 'msg': ...}]`) and then iterate over them or to start download video at this point